logstash-core/src/main/java/org/logstash/ackedqueue/Queue.java

```diff
@@ -290,7 +300,11 @@ public long write(Queueable element) throws IOException {
     public boolean isFull() {
         // TODO: I am not sure if having unreadCount as volatile is sufficient here. all unreadCount updates are done inside syncronized
         // TODO: sections, I believe that to only read the value here, having it as volatile is sufficient?
-        return (this.maxUnread > 0) ? this.unreadCount >= this.maxUnread : false;
+        if ((this.maxSizeInBytes > 0) && this.currentSize >= this.maxSizeInBytes) {
+            return true;
```

Comment:
Can we log debug here which `full` criteria was satisfied? We also need metrics, but I think @talevy is looking at it.